backend/app/Services/Models/ItemService.php

```diff
@@ -42,25 +47,84 @@ public function __construct(
      */
     public function create($data)
     {
-      $item = $this->model->store($data);
+      $data = $this->makeDataComplete($data);
 
-      $this->storage->downloadPoster($item->poster);
+      $item = $this->model->store($data);
 
       $this->episodeService->create($item);
-
       $this->alternativeTitleService->create($item);
 
+      $this->storage->downloadPoster($item->poster);
+      $this->storage->downloadBackdrop($item->backdrop);
+
       return $item;
     }
 
     /**
+     * Search against TMDb and IMDb for more informations.
+     * We don't need to get more informations if we add the item from the subpage.
+     *
+     * @param $data
+     * @return array
+     */
+    public function makeDataComplete($data)
+    {
+      if( ! isset($data['imdb_id'])) {
+        $details = $this->tmdb->details($data['tmdb_id'], $data['media_type']);
+        $title = $details->name ?? $details->title;
+
+        $data['imdb_id'] = $data['imdb_id'] ?? $this->parseImdbId($details);
```

Comment:
Not sure how php handles references or if it uses a copy of the param, but is it save to change the values of $data? My concern is that if it is a reference than this could bring in side effects. Wouldn't it be better to use a (deep) copy of $data and modify that instead?